pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -71,6 +71,8 @@ func (s *Service) ReconcileLoadBalancer(openStackCluster *infrav1.OpenStackClust
 		if v.Name == defaultLoadBalancerProvider {
 			lbProvider = v.Name
 			break
+		} else {
+			lbProvider = openStackCluster.Spec.APIServerLoadBalancer.Provider
```

Comment:
this will still choose amphora? 
should we check whether it's enabled in backend then use it (otherwise error out)
if not given, then use default one?